doc/reference/std/debug.md

```diff
@@ -1,190 +1,386 @@
+```
+> (dump-heap-stats!)
+=== memory usage ===
+gc-heap-size: 31118680
+gc-alloc: 72143824
+gc-live: 12403320
+gc-movable: 11693344
+gc-still: 709976
+=== heap summary ===
+objects: 185028
+still: 73
+refcounted: 1
+=== heap type counts ===
+pair 59812
+gx#import-binding::t 26592
+gx#module-import::t 23607
+string 17595
+symbol 16772
+gx#syntax-quote::t 11961
+gx#module-export::t 7968
+procedure 7476
+vector 6542
+gx#module-binding::t 2065
+gx#extern-binding::t 1551
+gx#syntax-binding::t 523
+gx#macro-expander::t 357
+bignum 311
+##type-6-F3F63A41-2974-4D41-8B24-1744E866741D 251
+weak 226
+##type-5 166
+keyword 164
+gx#import-set::t 126
+foreign 111
+promise 103
+gx#export-set::t 81
+gx#context-phi::t 62
+gx#module-context::t 58
+gx#private-feature-expander::t 58
+gerbil.core#runtime-struct-exhibitor::t 49
+gerbil.core#extended-stuct-info::t 49
+std/generic/macros#primitive-type-info::t 36
+gerbil.core#extended-class-info::t 27
+gerbil.core#runtime-class-exhibitor::t 27
+std/generic/macros#builtin-type-info::t 21
+gx#expander-mark::t 21
+##type-11-42fe9aac-e9c6-4227-893e-a0ad76f58932 19
+gx#alias-binding::t 19
+gx#rename-macro-expander::t 18
+box 18
+gx#expression-form::t 15
+gerbil/runtime/eval#__core-special-form::t 13
+gerbil/runtime/eval#__core-expression::t 12
+gerbil.core#runtime-class-info::t 10
+gx#core-macro::t 10
+u8vector 9
+gx#reserved-expander::t 8
+gerbil#AST::t 8
+gx#top-special-form::t 8
+gx#feature-expander::t 7
+u16vector 6
+##type-11-6bd864f0-27ec-4639-8044-cf7c0135d716 6
+gx#export-expander::t 5
+gx#import-expander::t 5
+##type-49-a1d146d0-78f6-437f-aa67-3b9b5bb333dc 4
+s32vector 4
+f64vector 4
+gx#definition-form::t 4
+##type-41-B1E5C9EE-B4F9-4D3C-AACD-FE12D696101F 4
+gerbil/runtime/eval#__core-form::t 4
+gx#prelude-context::t 3
+frame 2
+##type-2-EACE4200-EB43-4891-852A-03C7B5D385E4 2
+gx#import-export-expander::t 2
+gx#module-special-form::t 2
+unknown 2
+gx#top-binding::t 2
+continuation 2
+gx#special-form::t 2
+##type-7-cd5f5bad-f96f-438d-8d63-ff887b7b39de 2
+##type-15-713f0ba8-1d76-4a68-8dfa-eaebd4aef1e3 2
+gerbil/runtime/eval#__context::t 2
+std/generic/dispatch#generic-table::t 1
+std/generic/dispatch#generic::t 1
+std/generic/macros#generic-info::t 1
+##type-34-d05e0aa7-e235-441d-aa41-c1ac02065460 1
+flonum 1
+std/interface#interface-info::t 1
+std/interface#interface-descriptor::t 1
+gx#top-context::t 1
+##type-8-1b002758-f900-4e96-be5e-fa407e331fc0 1
+tmp/benchmark/interface/interface-lib#Operation::t 1
+std/iter#:iter-end::t 1
+tmp/benchmark/interface/interface-lib#LinearAccumulator::t 1
+gx#root-context::t 1
+##type-21-A6899D11-290C-42A6-B47A-57C6B908698F 1
+##type-17-D7D5784C-16DC-4453-8832-CC05A6FE9353 1
 ```
-:::
-
-Please document me!
 
 ### walk-heap!
-::: tip usage
 ```
-(walk-heap! ...)
+(walk-heap! walk: (walk #f) root: (root #f)) -> hash-table
 ```
-:::
 
-Please document me!
+Walks the heap, optionally starting from the specified `root` object
+and using `walk` as the function for walking containers.
```

Comment:
What are containers?

Reply:
objects that contain other objects, i'll clarify.